### PR TITLE
MAINT: deprecate PdfMerger,AnnotationBuilder and other depreciations cleanup

### PR DESCRIPTION
**Pull request by pubpub-zz**
in preparation of release 5.0.0 coming

### CODE REVIEW

**Comment by pubpub-zz** [comment]
need first to improve coverage (indirectly degraded)

**Comment by pubpub-zz** [comment]
to be able to pass coverage, I add to reactivate some code that is currently unreachable (outlines currently can only word with dest array) issue #2814 is here to track this

**Comment by pubpub-zz** [comment]
This PR is ready for review: the coverage reduced by 0.05% comes from the code size decreased

**Comment by pubpub-zz** [comment]
@stefan6419846 
this PR is ready for review

File: pypdf/_writer.py [hunk 1]
```diff
@@ -1806,14 +1806,15 @@ def add_outline_item_dict(
         outline_item_object = TreeObject()
         outline_item_object.update(outline_item)
 
+        """code currently unreachable
```
Comment (stefan6419846) [inline pypdf/_writer.py:1809]: This is wrong as this still is a public method.

File: pypdf/_writer.py [hunk 2]
```diff
@@ -2500,7 +2501,7 @@ def _create_stream(
             stream = BytesIO(filecontent)
         else:
             raise NotImplementedError(
-                "PdfMerger.merge requires an object that PdfReader can parse. "
+                "merge requires an object that PdfReader can parse. "
```
Comment (stefan6419846) [inline pypdf/_writer.py:2504]:
```suggestion
                "Merging requires an object that PdfReader can parse. "
```

File: pypdf/constants.py [hunk 3]
```diff
@@ -210,7 +168,7 @@ class PagesAttributes:
     PARENT = "/Parent"  # dictionary, required; indirect reference to pages object
     KIDS = "/Kids"  # array, required; List of indirect references
     COUNT = "/Count"  # integer, required; the number of leaf nodes (page objects)
-                      # that are descendants of this node within the page tree
+    # that are descendants of this node within the page tree
```
Comment (stefan6419846) [inline pypdf/constants.py:171]:
```suggestion
                     # that are descendants of this node within the page tree
```

**Comment by pubpub-zz** [comment]
@stefan6419846 
good to complete

File: pypdf/_writer.py (hunk 1 above)
Comment (stefan6419846) [reply pypdf/_writer.py:1809]: Could you please elaborate this before resolving?

File: pypdf/constants.py [hunk 4]
```diff
@@ -210,7 +168,7 @@ class PagesAttributes:
     PARENT = "/Parent"  # dictionary, required; indirect reference to pages object
     KIDS = "/Kids"  # array, required; List of indirect references
     COUNT = "/Count"  # integer, required; the number of leaf nodes (page objects)
-                      # that are descendants of this node within the page tree
+                     # that are descendants of this node within the page tree
```
Comment (stefan6419846) [inline pypdf/constants.py:171]:
```suggestion
                      # that are descendants of this node within the page tree
```

File: pypdf/generic/_data_structures.py [hunk 5]
```diff
@@ -1041,7 +1030,7 @@ def get_data(self) -> bytes:
             return decoded.get_data()
 
     # This overrides the parent method:
-    def set_data(self, data: bytes) -> None:  # deprecated
+    def set_data(self, data: bytes) -> None:  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (stefan6419846) [reply pypdf/generic/_data_structures.py:1033]: Could you please elaborate this before resolving?

**Comment by pubpub-zz** [comment]
@stefan6419846 
Good now

**Review by stefan6419846 — APPROVED** [review]

**Pull request merged**